paasta_tools/api/views/instance.py

```diff
@@ -0,0 +1,132 @@
+# limitations under the License.
+"""
+PaaSTA service instance status/start/stop etc.
+"""
+import json
+import logging
+import traceback
+
+from pyramid.response import Response
+from pyramid.view import view_config
+
+from paasta_tools import marathon_tools
+from paasta_tools.api import settings
+from paasta_tools.cli.cmds.status import get_actual_deployments
+from paasta_tools.marathon_serviceinit import compose_marathon_job_instance_count
+from paasta_tools.marathon_serviceinit import deploy_status_marathon_job
+from paasta_tools.marathon_serviceinit import get_bouncing_status
+from paasta_tools.utils import NoDockerImageError
+from paasta_tools.utils import PaastaColors
+from paasta_tools.utils import validate_service_instance
+
+log = logging.getLogger(__name__)
+
+
+def chronos_instance_status(instance_status, service, instance, verbose):
+    return
+
+
+def marathon_job_status(client, job_config):
+    mstatus = {}
+
+    try:
+        app_id = job_config.format_marathon_app_dict()['id']
+        mstatus['app_id'] = app_id
+    except NoDockerImageError:
+        error_msg = "Docker image is not in deployments.json. Has Jenkins deployed it?"
+        mstatus['message'] = error_msg
+        return mstatus
+
+    if marathon_tools.is_app_id_running(app_id, client):
+        app = client.get_app(app_id)
+        mstatus['deploy_status'] = PaastaColors.decolor(
+            deploy_status_marathon_job(app, app_id, client))
+
+        running_instances = app.tasks_running
+        mstatus['running_instances'] = running_instances
+        normal_instance_count = job_config.get_instances()
+        mstatus['normal_instance_count'] = normal_instance_count
+
+        status, instance_count = compose_marathon_job_instance_count(
+            running_instances, normal_instance_count)
+        mstatus['status'] = PaastaColors.decolor(status)
+
+    else:
+        mstatus['status'] = 'Critical'
+        mstatus['deploy_status'] = 'Not Running'
+
+    return mstatus
+
+
+def marathon_instance_status(instance_status, service, instance, verbose):
+    marathon_config = marathon_tools.load_marathon_config()
+    client = marathon_tools.get_marathon_client(marathon_config.get_url(),
+                                                marathon_config.get_username(),
+                                                marathon_config.get_password())
+    job_config = marathon_tools.load_marathon_service_config(
+        service, instance, settings.cluster, soa_dir=settings.soa_dir)
+
+    instance_status['state'] = PaastaColors.decolor(
+        get_bouncing_status(service, instance, client, job_config))
+    instance_status['desired state'] = PaastaColors.decolor(
+        job_config.get_desired_state_human())
+    instance_status['marathon'] = marathon_job_status(client, job_config)
+
+
+def error_wrapper(error_message, error_code):
+    log.error(error_message)
+
+    response = Response(
+        body=json.dumps({'message': error_message}),
```

Comment:
what I'd really like here is for this to be a list of errors, rather than a single error message. It'd be much better if we can accumulate as much information for the caller at once, rather than the sequence of 'try one thing, get one error', 'try again, get another error'. It'd be better if we could return *all* the errors at once where it makes sense to do so.